src/app/Http/Controllers/Operations/ListOperation.php

```diff
@@ -113,18 +113,12 @@ public function search()
         // if there was no order set, this will be the only one
         // if there was an order set, this will be the last one (after all others were applied)
         $orderBy = $this->crud->query->getQuery()->orders;
-        $hasOrderByPrimaryKey = false;
-        collect($orderBy)->each(function ($item, $key) use ($hasOrderByPrimaryKey) {
-            if (! isset($item['column'])) {
-                return false;
-            }
-
-            if ($item['column'] == $this->crud->model->getKeyName()) {
-                $hasOrderByPrimaryKey = true;
+        $modelKey = $this->crud->model->getKeyName();
 
-                return false;
-            }
+        $hasOrderByPrimaryKey = collect($orderBy)->some(function ($item) use ($modelKey) {
+            return isset($item['column']) && $item['column'] === $modelKey || isset($item['sql']) && str_contains($item['sql'], ".$modelKey ");
```

Comment:
This does not account if you do `$this->crud->query->orderByRaw('id','asc')` it expects `table.id`.

Also what if I `orderByRaw('other_joined_table.id', 'desc')` it will assume that is this table order by primary key that is set ?

Reply:
@pxpm just pushed a commit to fix the joined tables issue.

Tested with;
- [x] `$this->crud->query->orderByRaw('table.id')`
- [x] `$this->crud->query->orderByRaw('joined_table.id')`
- [x] `$this->crud->query->orderBy('id');`

It fails on `$this->crud->query->orderByRaw('id')`, anyway, this function exists to add a simple improvement to the query, I think we shouldn't add more complexity here, to test cases like:
- `orderByRaw('id')` or 
- `orderByRaw('ideology asc')` or 
- `orderByRaw('joined.id asc, table.id desc')` or 
- `orderByRaw('joined.id, table.ideology')`

The possible cases are a lot, and we would need to carefully parse the raw sql. 
Long story short, the current code covers >95% of the use cases, the other 5% are not even affect by this.

Let me know what you think about this ✌

Reply:
100% agree.